pkg/server/environment_create.go

```diff
@@ -47,14 +50,26 @@ func (s *Server) environmentCreate(c *gin.Context) {
 		consts.LabelUID:             it,
 		consts.LabelEnvironmentName: req.Name,
 	}
+
+	logrus.WithFields(logrus.Fields{
+		"identity_token": it,
+		"image_labels":   cfg.Labels,
+		"environment":    req.Environment,
+	}).Debug("creating the environment")
 	annotations := map[string]string{}
 	for k, v := range cfg.Labels {
 		annotations[k] = v
 	}
 
+	ports, err := imageutil.PortsFromLabel(cfg.Labels[consts.ImageLabelPorts])
+	if err != nil {
+		c.JSON(500, errors.Wrap(err, "failed to get ports from label"))
+		return
+	}
+
 	hostKeyPath := "/var/envd/hostkey"
 	authKeyPath := "/var/envd/authkey"
-	var defaultPermMode int32 = 0600
+	var defaultPermMode int32 = 0666
```

Comment:
Shall we make this a const?

Reply:
Let's postpone it since the code may change frequently.